lib/devices/Roborock.js

```diff
@@ -335,18 +344,67 @@ class Roborock extends MiioVacuum {
         await this.sendLocal("set_timezone", [new_zone], {});
     }
 
-    /*
-        0-100: percent
-
-        Or presets:
-        101: quiet
-        102: balanced
-        103: Turbo
-        104: Max
-        105: Mop
+    /**
+     * @private
+     */
+    detectFanSpeeds(msg_ver) {
+        let fanSpeeds = {
+            "whisper": {label: "Whisper", value: 1},
+            "min": {label: "Quiet", value: 38},
+            "medium": {label: "Balanced", value: 60},
+            "high": {label: "Turbo", value: 75},
+            "max": {label: "Max", value: 100},
+            "mop": {label: "Mop", value: 105},
+        };
+
+        if (msg_ver >= 3) {
+            delete fanSpeeds.whisper;
+            fanSpeeds.min.value = 101;
+            fanSpeeds.medium.value = 102;
+            fanSpeeds.high.value = 103;
+            fanSpeeds.max.value = 104;
+            fanSpeeds.mop.value = 105;
```

Comment:
Should use quotes consistently, e.g.

    delete fanSpeeds["whisper"];
    fanSpeeds["min"].value = …

Reply:
fixed